Add color and typography styles on the mini cart title block

daniloparrajr opened the pull request:
The goal of this PR is to make the Mini Cart Title customizable through the settings sidebar (colors, typography).
In which the two inner blocks `Mini Cart Title Label` and `Mini Cart Title Items Counter` will inherit by default.

Fixes #9346

### Testing

1. Install and activate the Twenty Twenty-Three theme
1. Go to Appearance > Editor > Template parts > Mini Cart
1. Select the `Mini Cart Title` block and add some customizations on the settings sidebar (colors, typography).
1. Check if the customizations are applied in the editor and in the frontend of the website.

### Screenshots

<table>
<tr>
<td valign="top">Before:
<br><br>

 ![image](https://user-images.githubusercontent.com/25006926/236684541-f60af3bd-294d-4b48-8529-a7c7576691a0.png)
</td>
<td valign="top">After:
<br><br>

![image](https://user-images.githubusercontent.com/25006926/236684549-d38e2c98-c85a-467c-94de-0738457da314.png)
</td>
</tr>
</table>

### WooCommerce Visibility

* [X] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Performance Impact

<!-- Please document any known performance impact (positive or negative) here. If negative, provide some rationale for why this is an okay tradeoff or how this will be addressed. -->

### Changelog

> Mini cart: add color and typography styles on the mini cart title block

daniloparrajr committed:
feat: added color and typography supports on the mini cart title block

Aljullu reviewed (commented):
This is looking great, @daniloparrajr, thank you so much for contributing with this PR.

It's testing as expected and code changes look good. My only comment would be that I think we should disable support for background color (sorry, I know that wasn't specified in the original issue :pray: ). The reason is that I feel it looks a bit off because it has some spacing around it (padding and margin from other elements):

![imatge](https://user-images.githubusercontent.com/3616980/236764371-e06a74d2-4e7f-419a-8ead-02092d31e1f5.png)

For now, I think being able to change the text color and font size would be enough.

nielslange commented:
@Aljullu How did you manage to see the background color? On my end, both when using TT2 and TT3, I do not see the background color, as the following SCSS definition overwrites that setting on my end:

https://github.com/woocommerce/woocommerce-blocks/blob/7f8452ab130e8db2c5e78f2eb563b1ea6f5e8279/assets/js/blocks/mini-cart/style.scss#L126

<img width="1512" alt="Screenshot 2023-05-08 at 15 43 35" src="https://user-images.githubusercontent.com/3323310/236778591-c8c50687-0c80-4f0b-a45d-e50c9ef791ad.png">

nielslange reviewed (dismissed):
Thanks for working on this, @daniloparrajr.

Besides @Aljullu [feedback](https://github.com/woocommerce/woocommerce-blocks/pull/9382#pullrequestreview-1416288141), could you add testing steps to the PR? In https://github.com/woocommerce/woocommerce-blocks/pull/8905, you can find a great example of testing steps. Adding them, does not only help for testing the PR now, but also later when it gets merged into core. 😀

daniloparrajr committed:
refactor: remove background support for mini cart title block

daniloparrajr commented:
Thank you @Aljullu and @nielslange for checking my PR, I added the change to disable the background color support for the mini cart title block. I also added some testing steps, I hope this is helpful.

Aljullu reviewed (approved):
> Thank you @Aljullu and @nielslange for checking my PR, I added the change to disable the background color support for the mini cart title block. I also added some testing steps, I hope this is helpful.

Thanks for updating the PR and adding testing steps, @daniloparrajr. This is looking great. I will wait until tests pass and after that I will proceed merging. It there aren't any followup issues, your change will be included in WC Blocks 10.3 and WC core WC 7.9. :tada: 

> How did you manage to see the background color? On my end, both when using TT2 and TT3, I do not see the background color, as the following SCSS definition overwrites that setting on my end:

@nielslange Which color did you choose? When choosing one from the palette, it comes with an `!important` tag in the CSS, so it would override the default background. But maybe custom colors don't? Sidenote: this shouldn't block this PR as we have removed support for background colors.
